### DISCUSSION TITLE
The Great Feedback Gathering

### DISCUSSION THREAD

**Discussion by ernado** [gotd/td#1352] [General]
Hi! 

I've decided to gather some feedback from users on gotd project.
Feel free to post your problems, issues, feature requests here.

**Comment by evan-sm** [comment]
* would love to see more code examples for different use cases
* gotd requires a steep learning curve because of its low-lvl nature, more sugar, helpers and high-level functions are appreciated. things like accessHash should be handled automatically just like telethon does
* zap logging is an unnecessary dependency, since log/slog package were introduced, make zap go in favor of slog

**Reply by celestix** [reply]
Agreed on the last one, probably introducing a logger interface would be a more flexible option to allow users use a logger of their choice.

**Reply by ernado** [reply]
Now logger is swappable. 
Also I've updated the docs: https://gotd.dev/

Some access hash issues were resolved too.

Thank you for feedback.

**Comment by dwjgroup** [comment]
I don’t know why, but my account is blocked as soon as I log in. The api_id I use is 2024, WInClient.

**Reply by dwjgroup** [reply]
It's really hard to use, the blocking is too strong, normal operation will be blocked. Use this https://github.com/LonamiWebs/Telethon Seals are low.

**Reply by ernado** [reply]
https://github.com/LonamiWebs/Telethon/issues/824

> When using the library, the account is immediately banished

**Reply by ernado** [reply]
Also if you use reverse-engineered oficial api_id's (like 2024), you are almost guaranteed to be banned, yes.

However I should investigate ban rate of telethon vs gotd and what clients are doing to trigger the ban, but this will require a lot of new accounts and time.

I've got some funding via https://polar.sh/gotd and https://t.me/gotd_dev (TON and USDT are supported), so will start doing it soon.

**Comment by ernado** [comment]
Further feeback could be posted directly as new discussion.
